### PR TITLE
Shayan/feq 2275/add skeleton loader to header

### PR DESCRIPTION
**Pull request by shayan-deriv**
changes:
- added skeleton loader and updated `AccountSwitcher/>` accordingly
- fixed the oauth redirect  issue when there was already a cookie with invalid token

### CODE REVIEW

File: src/components/AppHeader/AppHeader.tsx [hunk 1]
```diff
@@ -27,7 +30,9 @@ const AppHeader = () => {
                 <MenuItems />
             </Wrapper>
             <Wrapper variant='right'>
-                {activeLoginid ? (
+                {!activeAccount ? (
```
Comment (ameerul-deriv) [inline src/components/AppHeader/AppHeader.tsx:33]: can we resolve this nested ternary 👀

**Review by ameerul-deriv — APPROVED** [review]

File: src/components/AppHeader/AppHeader.tsx [hunk 2]
```diff
@@ -9,15 +10,61 @@ import { MenuItems } from './MenuItems';
...
+        }
+
+        if (activeLoginid) {
+            return (
+                <>
+                    <Notifications />
+                    {isDesktop && (
+                        <TooltipMenuIcon
+                            as='a'
+                            className='pr-3 border-r-[1px] h-[32px]'
```
Comment (nada-deriv) [inline src/components/AppHeader/AppHeader.tsx:36]: shouldn't the values be in rem?

File: src/components/AppHeader/SkeletonLoader/index.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
```
Comment (nada-deriv) [inline src/components/AppHeader/SkeletonLoader/index.tsx:1]:
```suggestion
import { Fragment } from 'react';
```

File: src/components/AppHeader/SkeletonLoader/index.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import ContentLoader from 'react-content-loader';
+
+type AccountsInfoLoaderProps = {
+    is_logged_in: boolean;
+    is_mobile: boolean;
+    speed: number;
```
Comment (nada-deriv) [inline src/components/AppHeader/SkeletonLoader/index.tsx:7]:
```suggestion
    ```suggestion
    isLoggedIn: boolean;
    isMobile: boolean;
    speed: number
```

File: src/components/AppHeader/SkeletonLoader/index.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+};
+
```
Comment (nada-deriv) [inline src/components/AppHeader/SkeletonLoader/index.tsx:9]:
```suggestion
type AccountsInfoLoaderProps = {
    isLoggedIn: boolean;
    isMobile: boolean;
    speed: number;
};
```

File: src/components/AppHeader/SkeletonLoader/index.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+const LoggedInPreloader = ({ is_mobile }: Pick<AccountsInfoLoaderProps, 'is_mobile'>) => (
```
Comment (nada-deriv) [inline src/components/AppHeader/SkeletonLoader/index.tsx:10]:
```suggestion
const LoggedInPreloader = ({ isMobile }: Pick<AccountsInfoLoaderProps, 'isMobile'>) => (
```

File: src/components/AppHeader/AccountSwitcher/index.tsx [hunk 7]
```diff
@@ -1,18 +1,22 @@
-import { api } from '@/hooks';
+import { useActiveAccount } from '@/hooks/api/account';
 import { CurrencyUsdIcon } from '@deriv/quill-icons';
 import { AccountSwitcher as UIAccountSwitcher } from '@deriv-com/ui';
 import { FormatUtils } from '@deriv-com/utils';
 
-export const AccountSwitcher = () => {
-    const { data } = api.account.useActiveAccount();
+type TActiveAccount = NonNullable<ReturnType<typeof useActiveAccount>['data']>;
+type AccountSwitcherProps = {
```
Comment (nada-deriv) [inline src/components/AppHeader/AccountSwitcher/index.tsx:7]:
```suggestion
type TAccountSwitcherProps = {
```

File: src/components/AppHeader/SkeletonLoader/index.tsx [hunk 8]
```diff
@@ -1,41 +1,41 @@
...
+            </>
         )}
     </>
 );
 
-export const AccountsInfoLoader = ({ is_mobile, speed }: AccountsInfoLoaderProps) => (
+export const AccountsInfoLoader = ({ isMobile, speed }: TAccountsInfoLoaderProps) => (
     <ContentLoader
         backgroundColor={'#f2f3f4'}
+        data-testid='accounts-info-loader'
```
Comment (nada-deriv) [inline src/components/AppHeader/SkeletonLoader/index.tsx:33]: data-testid='dt_accounts_info_loader'

File: src/components/AppHeader/AppHeader.tsx [hunk 9]
```diff
@@ -9,15 +12,63 @@ import { MenuItems } from './MenuItems';
...
     const { isDesktop } = useDevice();
+    const isEndpointPage = getCurrentRoute() === 'endpoint';
     const { activeLoginid, logout } = useAuthData();
+    const { data: activeAccount } = api.account.useActiveAccount();
     const { localize } = useTranslations();
     const oauthUrl = getOauthUrl();
 
+    const renderAccountSection = () => {
+        if (!isEndpointPage && !activeAccount) {
+            console.log('here:', isEndpointPage, activeAccount);
```
Comment (nada-deriv) [inline src/components/AppHeader/AppHeader.tsx:29]: is the console required?

Comment (shayan-deriv) [reply src/components/AppHeader/AppHeader.tsx]: good catch. removed it. thanks @nada-deriv

**Review by nada-deriv — APPROVED** [review]

**Pull request merged**